include/oneapi/dpl/pstl/utils.h

```diff
@@ -509,7 +498,7 @@ struct __is_pointer_to_const_member_impl<_R (_U::*)(_Args...) const noexcept> :
 {
 };
 
-template <typename _Tp, bool = __is_callable_object<_Tp>::value>
```

Comment:
It's a different semantics. I was looking in this trait for quite a while. I don't like how it's written (the whole family), though it was my suggestion is the past.

Clearly, `std::is_invocable` is not the same as `__callable_object`. The first one is much much wider. Please postpone this change unless we discuss the desirable design.

Reply:
Converted to 'draft'.
@rarutyun could you please demonstrate described difference on some example?

Reply:
@SergeyKopienko,

`std::is_invocable` relies on the well-formness of `INVOKE` expression (similar to `std::invoke`) that allows three possible types of callable objects:

* "Regular" callable objects (structures, lambdas, pointers to functions),
* Pointers to non-static functions
* Pointers to non-static variables

Consider [this example](https://godbolt.org/z/Gx89bK66c) for the last two usages.

Reply:
@kboyarinov but could you please describe the situations, for which we have the different behavior in this PR ?

Reply:
Sure, since you use `std::is_invocable_v` without providing any arguments, pointers to non-static functions/variables cannot affect the behavior.
But there can be a difference if the type provided is pointer to global function or static function. 
See [here](https://godbolt.org/z/MbjcP8xqb) for more details.

And since for `__is_callable_object` we want to check for exactly callable objects (structures or lambdas) - I guess we cannot use `std::is_invocable` here. 
I did not deeply investigate where __is_callable_object is used and can the logic be rewritten to avoid this trait.

Reply:
@kboyarinov Agree, as we can see in your example at https://godbolt.org/z/MbjcP8xqb, the behavior for pointers to global or static function really changed.
But I am sure that we should interpret this result as an error.

Reply:
OK, let's close this draft PR.